### PR TITLE
add day 16 AND 17

### PR DESCRIPTION
**Pull request by arora-aditya**
wasted sooooo much time on part 1 parsing oof

I was expecting part 2 to be harder and requiring combinations bur realized that the answer has to be deterministic so there would be only 1 way to solve this

### CODE REVIEW

**Comment by AngelOnFira** [comment]
My pt 1 was super fast, placed 143rd. Very slow pt 2 though, 1h+. Gonna do a lot of reading on others' solutions for this one :ok_hand:

File: 2020/day-16/arora-aditya/day16.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                break
+        if not flag:
+            invalids.append(num)
+    return invalids
+
+def part1(lines: List[str]) -> int:
+    i = 0
+    line = lines[i]
+    di = defaultdict(set)
+    while len(line) != 0:
```
Comment (AngelOnFira) [inline 2020/day-16/arora-aditya/day16.py:33]: Ya, I hardcoded my parsing. Def paid off for p1. Not great for testing the example though.

File: 2020/day-17/arora-aditya/day17.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    (0, -1, 0, 0),
+    (0, -1, 0, 1),
+    (0, -1, 1, -1),
+    (0, -1, 1, 0),
+    (0, -1, 1, 1),
+    (0, 0, -1, -1),
+    (0, 0, -1, 0),
+    (0, 0, -1, 1),
+    (0, 0, 0, -1),
+    # (0, 0, 0, 0),
```
Comment (arora-aditya) [inline 2020/day-17/arora-aditya/day17.py:85]: I memed myself so hard here smh

File: 2020/day-17/arora-aditya/day17.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    (1, -1, -1),
+    (1, -1, 0),
+    (1, -1, 1),
+    (1, 0, -1),
+    (1, 0, 0),
+    (1, 0, 1),
+    (1, 1, -1),
+    (1, 1, 0),
+    (1, 1, 1),
+]
```
Comment (AngelOnFira) [inline 2020/day-17/arora-aditya/day17.py:42]: Dunno if you generate these, but I made this on the first cellular automata of this year and just reused it:
```python
for xk in range(-1, 2):
    for yk in range(-1, 2):
        for zk in range(-1, 2):
            for wk in range(-1, 2):
                if xk == yk == zk == wk == 0:
                    continue
                ...
```

File: 2020/day-17/arora-aditya/day17.py (hunk 2 above)
Comment (AngelOnFira) [reply 2020/day-17/arora-aditya/day17.py:85]: LOL I didn't even know that you commented this cause I had to scroll down so far on the code file

File: 2020/day-17/arora-aditya/day17.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        for y, c in enumerate(l):
+            if c == '#':
+                g.add((x, y, 0))
+    return g
+
+def bounds(g):
+    res = []
+    for d in range(3):
+        lo = min(x[d] for x in g) - 1
+        hi = max(x[d] for x in g) + 2
```
Comment (AngelOnFira) [inline 2020/day-17/arora-aditya/day17.py:143]: Figuring out this -1 +2 was oof

**Comment by AngelOnFira** [comment]
Really solid time today, did you reuse the code from last year's 3D game of life? I'm mad that I didn't get a better template ready after we did it already this year :sweat_smile:

File: 2020/day-17/arora-aditya/day17.py (hunk 3 above)
Comment (arora-aditya) [reply 2020/day-17/arora-aditya/day17.py:42]: i had them from last year should've just generated

**Comment by arora-aditya** [comment]
yeah i did reuse

**Comment by AngelOnFira** [comment]
Ya, that would def allow for super small overhead of debugging potential. Having a more accessible library of solutions is what I should do I think

**Pull request merged**